tests/migration_test.go

```diff
@@ -469,6 +469,84 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 		return uid
 	}
 
+	runMigrationAndCollectMigrationMetrics := func(vmi *v1.VirtualMachineInstance, migration *v1.VirtualMachineInstanceMigration, timeout int) string {
+		var pod *k8sv1.Pod
+		var metricsIPs []string
+		var family k8sv1.IPFamily = k8sv1.IPv4Protocol
+
+		vmiNodeOrig := vmi.Status.NodeName
+		By("Finding the prometheus endpoint")
+		pod, err = kubecli.NewVirtHandlerClient(virtClient).Namespace(flags.KubeVirtInstallNamespace).ForNode(vmiNodeOrig).Pod()
+		Expect(err).ToNot(HaveOccurred(), "Should find the virt-handler pod")
+		for _, ip := range pod.Status.PodIPs {
+			metricsIPs = append(metricsIPs, ip.IP)
+		}
+
+		By("Starting a Migration")
+		Eventually(func() error {
+			migration, err = virtClient.VirtualMachineInstanceMigration(migration.Namespace).Create(migration, &metav1.CreateOptions{})
+			return err
+		}, timeout, 1*time.Second).ShouldNot(HaveOccurred())
+		By("Waiting until the Migration Completes")
+
+		if family == k8sv1.IPv6Protocol {
+			libnet.SkipWhenNotDualStackCluster(virtClient)
+		}
+
+		ip := getSupportedIP(metricsIPs, family)
+
+		By("Scraping the Prometheus endpoint")
+		var metrics map[string]float64
+		var lines []string
+
+		Eventually(func() map[string]float64 {
+			metricsURL := prepareMetricsURL(ip, 8443)
+			out, _, err := tests.ExecuteCommandOnPodV2(virtClient,
```

Comment:
Nice!
I see we're using the same logic [here](https://github.com/kubevirt/kubevirt/blob/main/tests/infra_test.go#L643). I think we better make it into an exported (a.k.a. "public") function and use it in both places, wdyt?

Reply:
tried it, it does not look nice to me, since you need to pass 2 more params to the function, and either make one that returns a function and use it in several spots in the infra_test or just pass the 2 extra params every time. And this is the only duplication, I tend to go with the rule of drying if 3 or more. But if you insist I can do that.

Reply:
aren't lines 503 to 513 identical to https://github.com/kubevirt/kubevirt/blob/main/tests/infra_test.go#L644-L654 ?

Reply:
added a lambda to remove code duplication.